File: binding/src/charts/libqcandlestickseries.cpp
#include <QAbstractSeries>
#include <QBrush>
#include <QCandlestickSeries>
#include <QCandlestickSet>
#include <QColor>
#include <QObject>
#include <QPen>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qcandlestickseries.h>
#include "libqcandlestickseries.h"
#include "libqcandlestickseries.hxx"

QCandlestickSeries* QCandlestickSeries_new() {
	 return new VirtualQCandlestickSeries();
}

QCandlestickSeries* QCandlestickSeries_new2(QObject* parent) {
	 return new VirtualQCandlestickSeries(parent);
}

libqt_string QCandlestickSeries_Tr(const char* s) {
	QString _ret = QCandlestickSeries::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

bool QCandlestickSeries_Append(QCandlestickSeries* self, QCandlestickSet* set) {
	return self->append(set);
}

bool QCandlestickSeries_Remove(QCandlestickSeries* self, QCandlestickSet* set) {
	return self->remove(set);
}

bool QCandlestickSeries_Append2(QCandlestickSeries* self, const libqt_list sets) {
	return self->append(*sets);
}

bool QCandlestickSeries_Remove2(QCandlestickSeries* self, const libqt_list sets) {
	return self->remove(*sets);
}

bool QCandlestickSeries_Insert(QCandlestickSeries* self, int index, QCandlestickSet* set) {
	return self->insert(index, set);
}

bool QCandlestickSeries_Take(QCandlestickSeries* self, QCandlestickSet* set) {
	return self->take(set);
}

void QCandlestickSeries_Clear(QCandlestickSeries* self) {
	self->clear();
}

libqt_list QCandlestickSeries_Sets(const QCandlestickSeries* self) {
	return self->sets();
}

int QCandlestickSeries_Count(const QCandlestickSeries* self) {
	return self->count();
}

int QCandlestickSeries_Type(const QCandlestickSeries* self) {
	return self->type();
}

void QCandlestickSeries_SetMaximumColumnWidth(QCandlestickSeries* self, double maximumColumnWidth) {
	self->setMaximumColumnWidth(maximumColumnWidth);
}

double QCandlestickSeries_MaximumColumnWidth(const QCandlestickSeries* self) {
	return self->maximumColumnWidth();
}

void QCandlestickSeries_SetMinimumColumnWidth(QCandlestickSeries* self, double minimumColumnWidth) {
	self->setMinimumColumnWidth(minimumColumnWidth);
}

double QCandlestickSeries_MinimumColumnWidth(const QCandlestickSeries* self) {
	return self->minimumColumnWidth();
}

void QCandlestickSeries_SetBodyWidth(QCandlestickSeries* self, double bodyWidth) {
	self->setBodyWidth(bodyWidth);
}

double QCandlestickSeries_BodyWidth(const QCandlestickSeries* self) {
	return self->bodyWidth();
}

void QCandlestickSeries_SetBodyOutlineVisible(QCandlestickSeries* self, bool bodyOutlineVisible) {
	self->setBodyOutlineVisible(bodyOutlineVisible);
}

bool QCandlestickSeries_BodyOutlineVisible(const QCandlestickSeries* self) {
	return self->bodyOutlineVisible();
}

void QCandlestickSeries_SetCapsWidth(QCandlestickSeries* self, double capsWidth) {
	self->setCapsWidth(capsWidth);
}

double QCandlestickSeries_CapsWidth(const QCandlestickSeries* self) {
	return self->capsWidth();
}

void QCandlestickSeries_SetCapsVisible(QCandlestickSeries* self, bool capsVisible) {
	self->setCapsVisible(capsVisible);
}

bool QCandlestickSeries_CapsVisible(const QCandlestickSeries* self) {
	return self->capsVisible();
}

void QCandlestickSeries_SetIncreasingColor(QCandlestickSeries* self, const QColor* increasingColor) {
	self->setIncreasingColor(*increasingColor);
}

QColor* QCandlestickSeries_IncreasingColor(const QCandlestickSeries* self) {
	return new QColor(self->increasingColor());
}

void QCandlestickSeries_SetDecreasingColor(QCandlestickSeries* self, const QColor* decreasingColor) {
	self->setDecreasingColor(*decreasingColor);
}

QColor* QCandlestickSeries_DecreasingColor(const QCandlestickSeries* self) {
	return new QColor(self->decreasingColor());
}

void QCandlestickSeries_SetBrush(QCandlestickSeries* self, const QBrush* brush) {
	self->setBrush(*brush);
}

QBrush* QCandlestickSeries_Brush(const QCandlestickSeries* self) {
	return new QBrush(self->brush());
}

void QCandlestickSeries_SetPen(QCandlestickSeries* self, const QPen* pen) {
	self->setPen(*pen);
}

QPen* QCandlestickSeries_Pen(const QCandlestickSeries* self) {
	return new QPen(self->pen());
}

void QCandlestickSeries_Clicked(QCandlestickSeries* self, QCandlestickSet* set) {
	self->clicked(set);
}

void QCandlestickSeries_Connect_Clicked(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*, QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSeries*, QCandlestickSet*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::clicked, [self, slotFunc](QCandlestickSet* set) {
	slotFunc(self, set);
    });
}

void QCandlestickSeries_Hovered(QCandlestickSeries* self, bool status, QCandlestickSet* set) {
	self->hovered(status, set);
}

void QCandlestickSeries_Connect_Hovered(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*, bool, QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSeries*, bool, QCandlestickSet*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::hovered, [self, slotFunc](bool status, QCandlestickSet* set) {
	slotFunc(self, status, set);
    });
}

void QCandlestickSeries_Pressed(QCandlestickSeries* self, QCandlestickSet* set) {
	self->pressed(set);
}

void QCandlestickSeries_Connect_Pressed(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*, QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSeries*, QCandlestickSet*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::pressed, [self, slotFunc](QCandlestickSet* set) {
	slotFunc(self, set);
    });
}

void QCandlestickSeries_Released(QCandlestickSeries* self, QCandlestickSet* set) {
	self->released(set);
}

void QCandlestickSeries_Connect_Released(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*, QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSeries*, QCandlestickSet*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::released, [self, slotFunc](QCandlestickSet* set) {
	slotFunc(self, set);
    });
}

void QCandlestickSeries_DoubleClicked(QCandlestickSeries* self, QCandlestickSet* set) {
	self->doubleClicked(set);
}

void QCandlestickSeries_Connect_DoubleClicked(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*, QCandlestickSet*) = reinterpret_cast<void (*)(QCandlestickSeries*, QCandlestickSet*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::doubleClicked, [self, slotFunc](QCandlestickSet* set) {
	slotFunc(self, set);
    });
}

void QCandlestickSeries_CandlestickSetsAdded(QCandlestickSeries* self, const libqt_list sets) {
	self->candlestickSetsAdded(*sets);
}

void QCandlestickSeries_Connect_CandlestickSetsAdded(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*, libqt_list) = reinterpret_cast<void (*)(QCandlestickSeries*, libqt_list)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::candlestickSetsAdded, [self, slotFunc](const QList<QCandlestickSet *>& sets) {
	slotFunc(self, sets);
    });
}

void QCandlestickSeries_CandlestickSetsRemoved(QCandlestickSeries* self, const libqt_list sets) {
	self->candlestickSetsRemoved(*sets);
}

void QCandlestickSeries_Connect_CandlestickSetsRemoved(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*, libqt_list) = reinterpret_cast<void (*)(QCandlestickSeries*, libqt_list)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::candlestickSetsRemoved, [self, slotFunc](const QList<QCandlestickSet *>& sets) {
	slotFunc(self, sets);
    });
}

void QCandlestickSeries_CountChanged(QCandlestickSeries* self) {
	self->countChanged();
}

void QCandlestickSeries_Connect_CountChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::countChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_MaximumColumnWidthChanged(QCandlestickSeries* self) {
	self->maximumColumnWidthChanged();
}

void QCandlestickSeries_Connect_MaximumColumnWidthChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::maximumColumnWidthChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_MinimumColumnWidthChanged(QCandlestickSeries* self) {
	self->minimumColumnWidthChanged();
}

void QCandlestickSeries_Connect_MinimumColumnWidthChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::minimumColumnWidthChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_BodyWidthChanged(QCandlestickSeries* self) {
	self->bodyWidthChanged();
}

void QCandlestickSeries_Connect_BodyWidthChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::bodyWidthChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_BodyOutlineVisibilityChanged(QCandlestickSeries* self) {
	self->bodyOutlineVisibilityChanged();
}

void QCandlestickSeries_Connect_BodyOutlineVisibilityChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::bodyOutlineVisibilityChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_CapsWidthChanged(QCandlestickSeries* self) {
	self->capsWidthChanged();
}

void QCandlestickSeries_Connect_CapsWidthChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::capsWidthChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_CapsVisibilityChanged(QCandlestickSeries* self) {
	self->capsVisibilityChanged();
}

void QCandlestickSeries_Connect_CapsVisibilityChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::capsVisibilityChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_IncreasingColorChanged(QCandlestickSeries* self) {
	self->increasingColorChanged();
}

void QCandlestickSeries_Connect_IncreasingColorChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::increasingColorChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_DecreasingColorChanged(QCandlestickSeries* self) {
	self->decreasingColorChanged();
}

void QCandlestickSeries_Connect_DecreasingColorChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::decreasingColorChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_BrushChanged(QCandlestickSeries* self) {
	self->brushChanged();
}

void QCandlestickSeries_Connect_BrushChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::brushChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QCandlestickSeries_PenChanged(QCandlestickSeries* self) {
	self->penChanged();
}

void QCandlestickSeries_Connect_PenChanged(QCandlestickSeries* self, intptr_t slot) {
    void (*slotFunc)(QCandlestickSeries*) = reinterpret_cast<void (*)(QCandlestickSeries*)>(slot);
    QCandlestickSeries::connect(self, &QCandlestickSeries::penChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QCandlestickSeries_Tr2(const char* s, const char* c) {
	QString _ret = QCandlestickSeries::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QCandlestickSeries_Tr3(const char* s, const char* c, int n) {
	QString _ret = QCandlestickSeries::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QCandlestickSeries_QBaseType(const QCandlestickSeries* self) {
	auto* vqcandlestickseries = dynamic_cast<const VirtualQCandlestickSeries*>(self);
	if (vqcandlestickseries && vqcandlestickseries->isVirtualQCandlestickSeries) {
vqcandlestickseries->setQCandlestickSeries_Type_IsBase(true);
	return vqcandlestickseries->type();
}
}

// Auxiliary method to allow providing re-implementation
void QCandlestickSeries_OnType(const QCandlestickSeries* self, intptr_t slot) {
	auto* vqcandlestickseries = dynamic_cast<const VirtualQCandlestickSeries*>(self);
	if (vqcandlestickseries && vqcandlestickseries->isVirtualQCandlestickSeries) {
vqcandlestickseries->setQCandlestickSeries_Type_Callback(reinterpret_cast<VirtualQCandlestickSeries::QCandlestickSeries_Type_Callback>(slot));
}
}

void QCandlestickSeries_Delete(QCandlestickSeries* self) {
    delete self;
}

File: binding/src/charts/libqcandlestickseries.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcandlestickseries.html
pub const qcandlestickseries = struct {

    /// New constructs a new QCandlestickSeries object.
    pub fn New() QtC.QCandlestickSeries {
        return qtc.QCandlestickSeries_new();
    }


    /// New2 constructs a new QCandlestickSeries object.
    pub fn New2(parent: ?*anyopaque) QtC.QCandlestickSeries {
        return qtc.QCandlestickSeries_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCandlestickSeries_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcandlestickseries.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Append(self: ?*anyopaque, set: ?*anyopaque) bool {
        return qtc.QCandlestickSeries_Append(@ptrCast(self), @ptrCast(set));
    }

    pub fn Remove(self: ?*anyopaque, set: ?*anyopaque) bool {
        return qtc.QCandlestickSeries_Remove(@ptrCast(self), @ptrCast(set));
    }

    pub fn Append2(self: ?*anyopaque, sets: []const u8) bool {
        return qtc.QCandlestickSeries_Append2(@ptrCast(self), @ptrCast(sets));
    }

    pub fn Remove2(self: ?*anyopaque, sets: []const u8) bool {
        return qtc.QCandlestickSeries_Remove2(@ptrCast(self), @ptrCast(sets));
    }

    pub fn Insert(self: ?*anyopaque, index: i32, set: ?*anyopaque) bool {
        return qtc.QCandlestickSeries_Insert(@ptrCast(self), index, @ptrCast(set));
    }

    pub fn Take(self: ?*anyopaque, set: ?*anyopaque) bool {
        return qtc.QCandlestickSeries_Take(@ptrCast(self), @ptrCast(set));
    }

    pub fn Clear(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_Clear(@ptrCast(self));
    }

    pub fn Sets(self: ?*anyopaque, ) []const u8 {
        return qtc.QCandlestickSeries_Sets(@ptrCast(self));
    }

    pub fn Count(self: ?*anyopaque, ) i32 {
        return qtc.QCandlestickSeries_Count(@ptrCast(self));
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QCandlestickSeries_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QCandlestickSeries_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QCandlestickSeries_QBaseType(@ptrCast(self));
    }

    pub fn SetMaximumColumnWidth(self: ?*anyopaque, maximumColumnWidth: f64) void {
        qtc.QCandlestickSeries_SetMaximumColumnWidth(@ptrCast(self), maximumColumnWidth);
    }

    pub fn MaximumColumnWidth(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSeries_MaximumColumnWidth(@ptrCast(self));
    }

    pub fn SetMinimumColumnWidth(self: ?*anyopaque, minimumColumnWidth: f64) void {
        qtc.QCandlestickSeries_SetMinimumColumnWidth(@ptrCast(self), minimumColumnWidth);
    }

    pub fn MinimumColumnWidth(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSeries_MinimumColumnWidth(@ptrCast(self));
    }

    pub fn SetBodyWidth(self: ?*anyopaque, bodyWidth: f64) void {
        qtc.QCandlestickSeries_SetBodyWidth(@ptrCast(self), bodyWidth);
    }

    pub fn BodyWidth(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSeries_BodyWidth(@ptrCast(self));
    }

    pub fn SetBodyOutlineVisible(self: ?*anyopaque, bodyOutlineVisible: bool) void {
        qtc.QCandlestickSeries_SetBodyOutlineVisible(@ptrCast(self), bodyOutlineVisible);
    }

    pub fn BodyOutlineVisible(self: ?*anyopaque, ) bool {
        return qtc.QCandlestickSeries_BodyOutlineVisible(@ptrCast(self));
    }

    pub fn SetCapsWidth(self: ?*anyopaque, capsWidth: f64) void {
        qtc.QCandlestickSeries_SetCapsWidth(@ptrCast(self), capsWidth);
    }

    pub fn CapsWidth(self: ?*anyopaque, ) f64 {
        return qtc.QCandlestickSeries_CapsWidth(@ptrCast(self));
    }

    pub fn SetCapsVisible(self: ?*anyopaque, capsVisible: bool) void {
        qtc.QCandlestickSeries_SetCapsVisible(@ptrCast(self), capsVisible);
    }

    pub fn CapsVisible(self: ?*anyopaque, ) bool {
        return qtc.QCandlestickSeries_CapsVisible(@ptrCast(self));
    }

    pub fn SetIncreasingColor(self: ?*anyopaque, increasingColor: ?*anyopaque) void {
        qtc.QCandlestickSeries_SetIncreasingColor(@ptrCast(self), @ptrCast(increasingColor));
    }

    pub fn IncreasingColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QCandlestickSeries_IncreasingColor(@ptrCast(self));
    }

    pub fn SetDecreasingColor(self: ?*anyopaque, decreasingColor: ?*anyopaque) void {
        qtc.QCandlestickSeries_SetDecreasingColor(@ptrCast(self), @ptrCast(decreasingColor));
    }

    pub fn DecreasingColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QCandlestickSeries_DecreasingColor(@ptrCast(self));
    }

    pub fn SetBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QCandlestickSeries_SetBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn Brush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QCandlestickSeries_Brush(@ptrCast(self));
    }

    pub fn SetPen(self: ?*anyopaque, pen: ?*anyopaque) void {
        qtc.QCandlestickSeries_SetPen(@ptrCast(self), @ptrCast(pen));
    }

    pub fn Pen(self: ?*anyopaque, ) QtC.QPen {
        return qtc.QCandlestickSeries_Pen(@ptrCast(self));
    }

    pub fn Clicked(self: ?*anyopaque, set: ?*anyopaque) void {
        qtc.QCandlestickSeries_Clicked(@ptrCast(self), @ptrCast(set));
    }

    pub fn OnClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_Clicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Hovered(self: ?*anyopaque, status: bool, set: ?*anyopaque) void {
        qtc.QCandlestickSeries_Hovered(@ptrCast(self), status, @ptrCast(set));
    }

    pub fn OnHovered(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool, ?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_Hovered(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Pressed(self: ?*anyopaque, set: ?*anyopaque) void {
        qtc.QCandlestickSeries_Pressed(@ptrCast(self), @ptrCast(set));
    }

    pub fn OnPressed(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_Pressed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Released(self: ?*anyopaque, set: ?*anyopaque) void {
        qtc.QCandlestickSeries_Released(@ptrCast(self), @ptrCast(set));
    }

    pub fn OnReleased(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_Released(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn DoubleClicked(self: ?*anyopaque, set: ?*anyopaque) void {
        qtc.QCandlestickSeries_DoubleClicked(@ptrCast(self), @ptrCast(set));
    }

    pub fn OnDoubleClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_DoubleClicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CandlestickSetsAdded(self: ?*anyopaque, sets: []const u8) void {
        qtc.QCandlestickSeries_CandlestickSetsAdded(@ptrCast(self), @ptrCast(sets));
    }

    pub fn OnCandlestickSetsAdded(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_CandlestickSetsAdded(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CandlestickSetsRemoved(self: ?*anyopaque, sets: []const u8) void {
        qtc.QCandlestickSeries_CandlestickSetsRemoved(@ptrCast(self), @ptrCast(sets));
    }

    pub fn OnCandlestickSetsRemoved(self: ?*anyopaque, callback: *const fn (?*anyopaque, [*:0]const u8) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_CandlestickSetsRemoved(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CountChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_CountChanged(@ptrCast(self));
    }

    pub fn OnCountChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_CountChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn MaximumColumnWidthChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_MaximumColumnWidthChanged(@ptrCast(self));
    }

    pub fn OnMaximumColumnWidthChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_MaximumColumnWidthChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn MinimumColumnWidthChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_MinimumColumnWidthChanged(@ptrCast(self));
    }

    pub fn OnMinimumColumnWidthChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_MinimumColumnWidthChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BodyWidthChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_BodyWidthChanged(@ptrCast(self));
    }

    pub fn OnBodyWidthChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_BodyWidthChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BodyOutlineVisibilityChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_BodyOutlineVisibilityChanged(@ptrCast(self));
    }

    pub fn OnBodyOutlineVisibilityChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_BodyOutlineVisibilityChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CapsWidthChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_CapsWidthChanged(@ptrCast(self));
    }

    pub fn OnCapsWidthChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_CapsWidthChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn CapsVisibilityChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_CapsVisibilityChanged(@ptrCast(self));
    }

    pub fn OnCapsVisibilityChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_CapsVisibilityChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn IncreasingColorChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_IncreasingColorChanged(@ptrCast(self));
    }

    pub fn OnIncreasingColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_IncreasingColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn DecreasingColorChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_DecreasingColorChanged(@ptrCast(self));
    }

    pub fn OnDecreasingColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_DecreasingColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BrushChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_BrushChanged(@ptrCast(self));
    }

    pub fn OnBrushChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_BrushChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PenChanged(self: ?*anyopaque, ) void {
        qtc.QCandlestickSeries_PenChanged(@ptrCast(self));
    }

    pub fn OnPenChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QCandlestickSeries_Connect_PenChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCandlestickSeries_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcandlestickseries.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QCandlestickSeries_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qcandlestickseries.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCandlestickSeries_Delete(@ptrCast(self));
    }
};
